src/Cli/dotnet/commands/dotnet-test/TestCommandParser.cs

```diff
@@ -78,7 +78,11 @@ internal static class TestCommandParser
         public static readonly Option<IEnumerable<string>> CollectOption = new ForwardedOption<IEnumerable<string>>("--collect", LocalizableStrings.cmdCollectDescription)
         {
             ArgumentHelpName = LocalizableStrings.cmdCollectFriendlyName
-        }.ForwardAsSingle(o => $"-property:VSTestCollect=\"{string.Join(";", o)}\"")
+        }.ForwardAsSingle(o =>
+        {
```

Comment:
can we just simply change it to:

}.ForwardAsSingle(o => $"-property:VSTestCollect=\"{string.Join(";", GetSemiColonEscapedArgs(o))}\"")
.Allow....

if you want to keep it longer please rename loggersString to collectString

Reply:
Updated.